File: src/main/java/com/github/njuro/jard/utils/TemplateService.java
package com.github.njuro.jard.utils;

import java.util.Map;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

@Service
public class TemplateService {

  private final TemplateEngine templateEngine;

  public TemplateService(TemplateEngine templateEngine) {
    this.templateEngine = templateEngine;
  }

  /**
   * Resolves template with given name and variables.
   *
   * @param template - name of the template (located in resources/templates - HTML file)
   * @param variables - map of template variables
   * @return resolved template
   */
  public String resolveTemplate(String template, Map<String, Object> variables) {
    var context = new Context();
    context.setVariables(variables);
    return templateEngine.process(template, context);
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/utils/TemplateService.kt
package com.github.njuro.jard.rewrite.utils

import org.springframework.stereotype.Service
import org.thymeleaf.TemplateEngine
import org.thymeleaf.context.Context

@Service
class TemplateService(private val templateEngine: TemplateEngine) {
    /**
     * Resolves template with given name and variables.
     *
     * @param template - name of the template (located in resources/templates - HTML file)
     * @param variables - map of template variables
     * @return resolved template
     */
    fun resolveTemplate(template: String, variables: Map<String, Any?>): String {
        val context = Context().apply { setVariables(variables) }
        return templateEngine.process(template, context)
    }
}
